crates/engine/schema/src/builder/graph/directives/composite/mod.rs

```diff
@@ -146,13 +132,31 @@ fn ingest_before_federation_directives<'sdl>(
     Ok(())
 }
 
-fn ingest_after_federation_directives<'sdl>(
-    ingester: &mut DirectivesIngester<'_, 'sdl>,
-    def: sdl::SdlDefinition<'sdl>,
-    dir: sdl::Directive<'sdl>,
+pub(crate) fn ingest_composite_field_directives_after_federation(
```

Comment:
Just a sidenote, if we are iterating over all the field definitions in multiple places, would it be possible to combine them all into one loop? I haven't checked if this is the case, just feels a bit expensive to iterate all the field definitions for one directive here.

Reply:
A few things need to be done in different phases. Some things could surely be optimized, but we'll see once we have a schema that takes a long time to build.